hoover_script.py

```diff
@@ -30,9 +30,167 @@
     raise RuntimeError("HOOVER_HOME environment variable is not set")
 home = Path(_home)
 
-def question(label, default):
-    rv = input("{} [{}]: ".format(label, default))
-    return rv.strip() or default
+interactive_mode = False
+
+class Param:
+    param_list = []
+
+    def __init__(self, name, default, environ=None, question_label=None, optional=False):
+        self.name = name
+        self.default = default
+        self.environ = environ
+        self.question_label = question_label
+        self.optional = optional
+        self.value = None
+        Param.param_list.append(self)
+
+    @staticmethod
+    def _question(label, default):
+        rv = input("{} [{}]: ".format(label, default))
+        return rv.strip() or default
+
+    def get(self):
```

Comment:
It's a good practice to resolve all configuration values before doing any actual work.